### PR TITLE
Gal-414 & Gal-315 - Calculate number to invite 

### PR DESCRIPTION
**Pull request by Abdul-NHS**
https://nhsd-jira.digital.nhs.uk/browse/GAL-414

Added calculate number to invite button in lsoaTable component which fires POST request to retrieve amount of people to invite

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/components/pagination.js [hunk 1]
```diff
@@ -1,104 +1,93 @@
...
             >
-              2
+              « Previous
             </a>
           </li>
-          <li class="nhsuk-footer__list-item" id="nextButton">
+          {paginationRange.map(pgNumber => (
+                    <li key={pgNumber}
+                        class="style_item__Y9BLA"
+                        /*class= {`style_current__K8c2u ${currentPage == pgNumber ? 'active' : ''} `} */
```
Comment (msandhuNHS) [inline src/app/components/pagination.js:68]: do we want to delete these comments?

**Review by msandhuNHS — CHANGES_REQUESTED** [review]

File: src/app/components/pagination.js (hunk 1 above)
Comment (Abdul-NHS) [reply src/app/components/pagination.js:68]: Hi Manny, left this in for the next pagination tickets, however will remove

**Review by msandhuNHS — APPROVED** [review]

**Review by andrew-cleveland — APPROVED** [review]

**Review by zainm97 — DISMISSED** [review]

**Review by Amar-NHS — DISMISSED** [review]
As discussed over, let's address those comments in next clean up ticket.

File: src/app/views/clinic_information/ClinicInformation.js [hunk 2]
```diff
@@ -119,14 +128,17 @@ class ClinicInformation extends Component {
     this.setState({
       appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetFillToInputValue / 100)),
     });
+    this.context.setState({
+      targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetFillToInputValue / 100))
+    })
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js]: will do in general refactor

File: src/app/views/clinic_information/ClinicInformation.js [hunk 3]
```diff
@@ -156,6 +207,9 @@ class ClinicInformation extends Component {
     this.setState({
       targetFillToInputValue: e.target.value,
     });
+    this.context.setState({
+      targetPercentageToFill: e.target.value
+    })
   }
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js]: will do in general refactor

File: src/app/views/clinic_information/ClinicInformation.js [hunk 4]
```diff
@@ -251,6 +305,9 @@ class ClinicInformation extends Component {
             this.setState({
               appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (this.state.targetFillToInputValue / 100)),
             });
+            this.context.setState({
+              targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (this.state.targetFillToInputValue / 100))
+            })
           })
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js:332]: will do in general refactor

File: src/app/views/clinic_information/ClinicInformation.js [hunk 5]
```diff
@@ -333,14 +397,18 @@ class ClinicInformation extends Component {
...
                     const targetPercentageValue = response.data.targetPercentage.N;
                     this.setState({
                       targetFillToInputValue: targetPercentageValue,
                       appsToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetPercentageValue / 100)),
                     });
+                    this.context.setState({
+                      targetAppToFill: Math.floor(this.context.state.recentInvitationHistory.appsRemaining * (targetPercentageValue / 100)),
+                      targetPercentageToFill: targetPercentageValue
+                    })
                   });
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js]: will do in general refactor

File: src/app/views/clinic_information/LsoaTable.jsx [hunk 6]
```diff
@@ -1,163 +1,221 @@
-import React from "react";
+import React, { useState } from "react";
+import Pagination from "@/app/components/pagination";
+import { AppStateContext } from '@/app/context/AppStateContext';
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/LsoaTable.jsx]: good spot

[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/LsoaTable.jsx]: sorr this has been actioned and is not there anymore

File: src/app/views/clinic_information/ClinicInformation.js [hunk 7]
```diff
@@ -259,12 +316,19 @@ class ClinicInformation extends Component {
   }
 
   componentDidMount() {
+    //Mocked the data below which is supposed to be retrieved from previous page - "Clinic Summary"
+    const icb = {
+      code: "QJK",
+      board: "NHS DEVON INTEGRATED CARE BOARD",
+      id: "NHS DEVON INTEGRATED CARE BOARD"
+    };
+
```
[the remark this answers is not in the document]
Comment (zainm97) [reply src/app/views/clinic_information/ClinicInformation.js:346]: done

**Review by zainm97 — APPROVED** [review]

**Review by Amar-NHS — APPROVED** [review]
Approved.

**Pull request merged**